src/logger/connectivityTask.c

```diff
@@ -373,6 +373,15 @@ void connectivityTask(void *params)
                 pr_info_str_msg(connParams->connectionName, ": timeout");
                 should_reconnect = true;
             }
+
+            /*
+             * If we didn't rx or tx a message, sleep for a bit so as not
+             * starve the other tasks.
+             */
+            if (pdFALSE == res && !msgReceived) {
```

Comment:
I don't see how this benefits. when there are no messages to receive, xQueueReceive will block, freeing up the scheduler to service other tasks. That's how FreeRTOS works with queues.

What problem are we trying to solve?